File: codes/week14/지원/35.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        n = len(nums)

        if target <= nums[0]:
            return 0 

        if target > nums[-1]:
            return n
        
        for i in range(1, n):
            if nums[i - 1] <= target and target <= nums[i]:
                return i

File: codes/week14/진현/35.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int binary_search(vector<int>& v, int sz, int tg) {
        int st = 0, ed = sz - 1;

        int mid;
        while (st <= ed) {
            mid = (st + ed) >> 1;

            if (v[mid] == tg)
                return mid;
            else if (v[mid] > tg)
                ed = mid - 1;
            else
                st = mid + 1;
        }

        return st;
    }

    int searchInsert(vector<int>& nums, int target) {
        return binary_search(nums, nums.size(), target);
    }
};